### PR TITLE
Docker compose for full infra

### PR DESCRIPTION
**Pull request by ddimaria**

### CODE REVIEW

File: quadratic-client/package.json [hunk 1]
```diff
@@ -86,12 +86,13 @@
...
     "build": "export VITE_VERSION=$(git rev-parse HEAD) && export NODE_OPTIONS=--max-old-space-size=16384 && vite build",
     "build:prod": "tsc ../quadratic-shared/*.ts && export VITE_VERSION=$GIT_COMMIT && export NODE_OPTIONS=--max-old-space-size=16384 && vite build",
+    "build:docker": "export VITE_VERSION=$(git rev-parse HEAD) && export NODE_OPTIONS=--max-old-space-size=16384 && vite build --mode docker",
     "build:wasm": "npm run build:wasm:javascript && npm run build:wasm:nodejs && npm run build:wasm:types",
     "build:wasm:types": "cd .. && cd quadratic-core && cargo run --bin export_types",
     "build:wasm:javascript": "cd .. && cd quadratic-core && wasm-pack build --target web --out-dir ../quadratic-client/src/app/quadratic-core --weak-refs",
     "build:wasm:nodejs": "cd .. && cd quadratic-core && wasm-pack build --target nodejs --out-dir ../quadratic-client/src/app/quadratic-core/__mocks__ --weak-refs",
     "build:wasm:javascript:dev": "cd .. && cd quadratic-core && wasm-pack build --target web --out-dir ../quadratic-client/src/app/quadratic-core --weak-refs --dev",
-    "start:production": "npm run build && serve build",
+    "start:production": "serve build",
```
Comment (ddimaria) [inline quadratic-client/package.json:95]: @davidkircos this is the only change I'm concerned about.  It's been a while, so I need to research why this was done (json doesn't support comments 😠)

Comment (ddimaria) [reply quadratic-client/package.json:95]: Convo reference: https://quadratichq.slack.com/archives/D05PWQN71AT/p1717107775525889

File: quadratic-api/tsconfig.production.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
+  "extends": "./tsconfig.json",
```
Comment (davidkircos) [inline quadratic-api/tsconfig.production.json:2]: is this needed?

File: docker-compose.yml [hunk 3]
```diff
@@ -128,6 +193,33 @@ services:
...
+  #     - "3003:3003"
+  #   depends_on:
+  #     postgres:
+  #       condition: service_healthy
+  #     quadratic-api:
+  #       condition: service_started
+  #   profiles:
+  #     - backend
+  #     - quadratic-connection
+  #     - all
```
Comment (davidkircos) [inline docker-compose.yml:222]: cleanup

**Review by davidkircos — APPROVED** [review]

**Review by davidkircos — CHANGES_REQUESTED** [review]
Rust tests fail

File: docker-compose.yml (hunk 3 above)
Comment (ddimaria) [reply docker-compose.yml]: Done

File: quadratic-api/tsconfig.production.json (hunk 2 above)
Comment (ddimaria) [reply quadratic-api/tsconfig.production.json:2]: It applies values from `tsconfig.json` to `tsconfig.production.json`, though additions in `tsconfig.production.json` override the extended values.

**Review by davidkircos — APPROVED** [review]
rust tests pass, approved!

**Pull request merged**